tspng/extraction.py

```diff
@@ -144,3 +149,18 @@ def extract_from_folder(path: str, mime_type: str=MIME_TYPE) -> Dict:
     if file_list == []:
         raise Exception(f"The {path} does not contain a PNG file.")
     return extract_from_files(file_list, mime_type)
+
+def extract_from_url(url: str, mime_type: str=MIME_TYPE) -> Dict:
+    '''
+    Returns the metadata from a TS byte stream as a dictionary.
+
+        Parameters:
+                url (str): URL to a TS PNG file
+                mime_type (str): Optional; Media type of file,
+                    default is 'application/vnd.theiascope.io+json'
+
+        Returns:
+                (dict): Dictionary containing file metadata
+    '''
+    urllib.request.urlretrieve(url,'url')
```

Comment:
This does not look right. I think you want to use `urlopen` and implement something similar to [this Stackoverflow Answer](https://stackoverflow.com/a/68804963), _but_ "write" the response to bytes/BytesIO buffer instead of a file.

Reply:
Changes made